tests/compile-test.rs

```diff
@@ -181,8 +181,11 @@ fn run_ui_cargo(config: &mut compiletest::Config) {
                 }
 
                 let src_path = case.path().join("src");
-                env::set_current_dir(&src_path)?;
+                if !src_path.exists() {
+                    continue;
+                }
```

Comment:
I'd like to have a small comment here why this is necessary (basically what you have in the PR description).